hcl/eval/partial.go

```diff
@@ -1119,6 +1260,70 @@ func isTmFuncall(tok *hclwrite.Token) bool {
 		strings.HasPrefix(string(tok.Bytes), "tm_")
 }
 
+func isSameTokens(a, b hclwrite.Tokens) bool {
+	//fmt.Printf("len(a): %d, len(b): %d: `%s` != `%s`\n", len(a), len(b), a.Bytes(), b.Bytes())
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := 0; i < len(a); i++ {
+		if b[i].Type != a[i].Type ||
+			string(b[i].Bytes) != string(a[i].Bytes) {
+			//fmt.Printf("a[%d] != b[%d]: `%s` != `%s`\n", i, i, a[i].Bytes, b[i].Bytes)
+			return false
+		}
+	}
+	//fmt.Printf("are the same\n")
+	return true
+}
+
+// pushfrom push the source and evaluated tokens from other node into this one.
+func (n *node) pushfrom(other *node) {
+	n.source = append(n.source, other.source...)
+	n.evaluated = append(n.evaluated, other.evaluated...)
+}
+
+// push the token into both source and evaluated.
+func (n *node) push(tok *hclwrite.Token) {
+	n.source = append(n.source, tok)
+	n.evaluated = append(n.evaluated, tok)
+}
+
+func (n *node) pushEvaluated(toks ...*hclwrite.Token) {
+	n.evaluated = append(n.evaluated, toks...)
+}
+
+func (n *node) pushSource(toks ...*hclwrite.Token) {
+	n.source = append(n.source, toks...)
+}
+
+type nodestack struct {
+	elems []*node
```

Comment:
```suggestion
	nodes []*node
```

Even though from context it can be infered that we have nodes, calling it nodes is the same amount of typing and don't seem to be worse than elems.